From 49ec5fa5d9bb939b80ef3994b5f0c04e7525ab9f Mon Sep 17 00:00:00 2001
From: Daniel-Constantin Mierla <miconda@gmail.com>
Date: Wed, 6 Apr 2016 19:22:58 +0200
Subject: [PATCH] core: common check for ifa->ifa_addr in add_intefaces

- if no ip address associated with the network interface, it is not
  useful anyhow

(cherry picked from commit c395ae00d7970be0a857265e9324a50bcb00f4d0)
---
 socket_info.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/socket_info.c b/socket_info.c
index 95c6ee6..aa8928a 100644
--- a/socket_info.c
+++ b/socket_info.c
@@ -1205,9 +1205,12 @@ int add_interfaces(char* if_name, int family, unsigned short port,
 
 	for (ifa = ifap; ifa; ifa = ifa->ifa_next)
 	{
+		/* skip if no IP addr associated with the interface */
+		if (ifa->ifa_addr==0)
+			continue;
 #ifdef AF_PACKET
 		/* skip AF_PACKET addr family since it is of no use later on */
-		if (ifa->ifa_addr && ifa->ifa_addr->sa_family == AF_PACKET)
+		if (ifa->ifa_addr->sa_family == AF_PACKET)
 			continue;
 #endif
 		if (if_name && strcmp(if_name, ifa->ifa_name))
-- 
2.8.0.rc3

